SoundCrate/scfront
# [Jem] Full Functionality: Lists

## Context
- Users can create song lists and add songs to those lists.

## Changes
- Created 6 API routes for different list functionalities
- Updated the "+" Button to showcase immediate changes to lists on the Song Page.
- Reflects the exact amount of lists a user has, how many songs within a list, and what songs are in each list.

## Testing
1. Login as JaneDoe (inf124@gmail.com)
2. View the profile page: http://localhost:3000/user/janedoe/profile/
3. Create a new list with a new title and a description. Check changes appear on MongoDB.
4. Visit a song (or multiple) via the Home page or by searching.
5. Add/Remove the song(s) to/from one or more Lists.
6. Ensure that the song count and the "Song In" List behind the modal reflects the immediate changes.
7. Ensure that changes are reflected throughout all list-related screens starting at the profile page.
8. Check MongoDB to confirm that lists are being updated accordingly. 

## ToDo
- Remove hardcoded user-ids for getListsForModal and getListsBySong and refactor with middleware.
- Refactor getSongLists to work for both itself and getListsForModal once TODO above is finished. Change AddToListModal route API and delete getListsForModal.

<img width="764" alt="Screenshot 2024-06-05 at 9 19 02 PM" src="https://github.com/SoundCrate/scfront/assets/93406637/c4b40a22-9067-4646-a5b6-84a01b67336b">
<img width="743" alt="Screenshot 2024-06-05 at 9 18 44 PM" src="https://github.com/SoundCrate/scfront/assets/93406637/307d4b38-50c9-406d-b2dd-fc6feebf0ed7">
<img width="758" alt="Screenshot 2024-06-05 at 9 18 35 PM" src="https://github.com/SoundCrate/scfront/assets/93406637/fd9d6a7d-12dd-4c50-b265-ff33e5d3b1a5">
<img width="754" alt="Screenshot 2024-06-05 at 9 18 17 PM" src="https://github.com/SoundCrate/scfront/assets/93406637/843653c2-547f-411b-8367-f4d0bc1c24ac">
<img width="566" alt="Screenshot 2024-06-05 at 9 17 57 PM" src="https://github.com/SoundCrate/scfront/assets/93406637/0df3f787-1899-45fc-b1ce-3ae0cbbcdb23">
<img width="783" alt="Screenshot 2024-06-05 at 9 17 52 PM" src="https://github.com/SoundCrate/scfront/assets/93406637/5cffe4c9-d9a9-4dce-9d66-7447e3e4dbb6">
